tools/MarkdownConverter/Spec/SourceLocation.cs

```diff
@@ -26,53 +27,92 @@ public string Description
     {
         get
         {
-            if (_loc != null)
-            {
-                return _loc;
-            }
+            ComputeLazyFields();
```

Comment:
The diff here is much less useful than I'd hope. Conceptually, here are the changes:

1. Add two new properties for the start and end line of the span so that errors are highlighted correctly.
2. Factor out a helper method to initialize the three private backing fields. It's effectively the same logic in all three cases.